src/mongo/crypto/crypto_openssl.cpp

```diff
@@ -39,25 +39,47 @@
 #include <openssl/hmac.h>
 #include <openssl/sha.h>
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#include <cstring>
+
+static void *OPENSSL_zalloc(size_t num)
+{
+   void *ret = OPENSSL_malloc(num);
+
+   if (ret != NULL)
+       memset(ret, 0, num);
+   return ret;
+}
+
+EVP_MD_CTX *EVP_MD_CTX_new(void)
+{
+   return (EVP_MD_CTX*) OPENSSL_zalloc(sizeof(EVP_MD_CTX));
```

Comment:
Style nit: when we return here, we should use static_cast, rather than use a C style cast.